### PR TITLE
Query Builder

### PR DESCRIPTION
**Pull request by Robert-W**
This PR includes the following:

* BaseBuilder class that the QueryBuilder extends. It has query methods, boolean methods, and aggregation methods with two build functions, one for boolean queries and one that returns the current aggregation object.
* QueryBuilder class with methods for from, size, raw, build, buildAggregation, buildDisMax, and buildMultiMatch.
* Comprehensive tests with 100% Coverage.
* Ability to nest queries easily by passing a callback to any building function (except for filteredAggs which is meant for adding facet counts to boolean queries).

Things that need to be added (these two items have issues opened already):

* Documentation on the README for getting started and API examples. 
* Support for function scoring and creating functions.

### CODE REVIEW

File: src/BaseBuilder.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	* @param {Function} nester - Function that allows for nesting queries
+	* @return {BaseBuilder} this
+	*/
+	must () {
+		saveQuery(BaseBuilder, this._queries, BOOL.MUST, ...arguments);
+		return this;
+	}
+
+	/**
+	* @description Add must boolean queries
```
Comment (jonterrylee) [inline src/BaseBuilder.js]: Update text to should

File: src/index.js [hunk 2]
```diff
@@ -1,6 +1,135 @@
...
+	* @return {QueryBuilder} this
+	*/
+	from (from) {
+		if (from !== undefined) { this._query.from = from; }
+		return this;
+	}
+
+	/**
+	* @description Update the size setting
+	* @param {number} newSize - New value for size
```
Comment (jonterrylee) [inline src/index.js]: Should just be size

File: src/BaseBuilder.js (hunk 1 above)
Comment (Robert-W) [inline src/BaseBuilder.js]: fixed

**Comment by jonterrylee** [comment]
Some misspellings but otherwise looks good.

**Pull request merged**